fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1499,8 +1499,18 @@ void ReadYourWritesTransaction::atomicOp( const KeyRef& key, const ValueRef& ope
 		throw used_during_commit();
 	}
 
-	if(key >= getMaxWriteKey())
+	if (key == metadataVersionKey) {
+		if(!tr.apiVersionAtLeast(610)) {
+			throw key_outside_legal_range();
```

Comment:
This should be checking whether we've exceeded the maximum write key before throwing (i.e. whether ACCESS_SYSTEM_KEYS is being used).

Reply:
I think this is still outstanding

Reply:
Although I guess I'm not sure what it should do if it was legitimately trying to write to this key in old versions

Reply:
I did change it, I added the && key >= getMaxWriteKey()